### PR TITLE
feat: Immediately disable/enable on optIn/optOut

### PR DESCRIPTION
**Pull request by tofumatt**
## Summary

Addresses issue #727. Converts all tracking code to rely on `window.googlesitekitTrackingEnabled`, so tracking is also instantly enabled/disabled.

## Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.4.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: assets/js/components/optin.js [hunk 1]
```diff
@@ -34,8 +34,8 @@ class Optin extends Component {
 		super( props );
 
 		this.state = {
-			scriptOnPage: !! googlesitekit.admin.trackingOptin,
-			optIn: !! googlesitekit.admin.trackingOptin,
+			scriptOnPage: !! global.googlesitekitTrackingEnabled,
+			optIn: !! global.googlesitekitTrackingEnabled,
```
Comment (felixarntz) [inline assets/js/components/optin.js]: @tofumatt Why not use `window.googlesitekitTrackingEnabled` in all these references like we're doing elsewhere?

Comment (aaemnnosttv) [reply assets/js/components/optin.js]: I'm guessing because `window` doesn't exist in Node, so this is necessary for Jest.

Comment (tofumatt) [reply assets/js/components/optin.js]: Actually, no real reason other than it's a bit more idiomatic for Webpack/multi-platform (node + browser, as @aaemnnosttv mentioned Jest is running in Node) code. I can change it to `window` and file an issue to move to `global` (which Webpack should translate to `window` anyway in the build), but it more clearly expresses the intent of what we're doing 🤷‍♂

I'll revert for now but file an issue 😄

**Comment by tofumatt** [comment]
I'm honestly unsure why this change caused the VRTs to change. It's possible that it wasn't reading the right setting before, causing it to be on by default in the screen being tested. There were references to an `Optin` variable instead of the proper `OptIn` variable too, so I'm thinking some of these parts weren't 100% properly hooked up in the past.

File: assets/js/components/optin.js (hunk 1 above)
Comment (tofumatt) [reply assets/js/components/optin.js]: (Filed #920, just for posterity.)

**Pull request merged**